components/infra-proxy-service/integration_test/nodes_test.go

```diff
@@ -0,0 +1,152 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/chef/automate/api/interservice/infra_proxy/request"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetNode(t *testing.T) {
+	ctx := context.Background()
+	t.Run("when the node exists, return the node successfully", func(t *testing.T) {
+		name := fmt.Sprintf("node-%d", time.Now().Nanosecond())
+
+		reqNode := &request.NodeObject{
+			ServerId: autoDeployedChefServerID,
+			OrgId:    autoDeployedChefOrganizationID,
+			Name:     name,
+		}
+
+		node, err := infraProxy.CreateNode(ctx, reqNode)
+		assert.NoError(t, err)
+		assert.NotNil(t, node)
+
+		req := &request.Node{
+			ServerId: autoDeployedChefServerID,
+			OrgId:    autoDeployedChefOrganizationID,
+			Name:     name,
+		}
+		res, err := infraProxy.GetNode(ctx, req)
+		assert.NoError(t, err)
+		assert.NotNil(t, res)
+		assert.Equal(t, res.Name, name)
+	})
+
+	t.Run("when the node does not exist, return node not found", func(t *testing.T) {
+		req := &request.Node{
+			ServerId: autoDeployedChefServerID,
+			OrgId:    autoDeployedChefOrganizationID,
+			Name:     fmt.Sprintf("no-node-%d", time.Now().Nanosecond()),
+		}
+
+		res, err := infraProxy.GetNode(ctx, req)
+		assert.Error(t, err)
+		assert.Nil(t, res)
+	})
+}
+
+func TestUpdateNodeTags(t *testing.T) {
```

Comment:
Additional tests suggested:
 * when attempting to delete a non-existent tag, leaves tag group unchanged (?? or throws an error, whatever is correct here)
 * when attempting to delete all tags, updates tag group to the empty list
 * when attempting to delete some existing and some non-existing tags simultaneously, does ??
 * when attempting to set tags to empty list, updates tag group to the empty list